src/Uno.UWP/Storage/StorageFile.saf.Android.cs

```diff
@@ -0,0 +1,88 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.App;
+using AndroidX.DocumentFile.Provider;
+using Uno.Storage.Internal;
+using Windows.Storage.FileProperties;
+using Windows.Storage.Streams;
+
+namespace Windows.Storage
+{
+	public partial class StorageFile
+	{
+		public static StorageFile GetFromSafDocument(DocumentFile document) =>
+			new StorageFile(new SafFile(document));
+
+		public static StorageFile GetFromSafUri(Android.Net.Uri uri) =>
+			new StorageFile(new SafFile(uri));
+
+		internal class SafFile : ImplementationBase
+		{
+			private readonly Android.Net.Uri _fileUri;
+			private readonly DocumentFile _directoryDocument;
+
+			internal SafFile(Android.Net.Uri uri)
+			{
+				_fileUri = uri ?? throw new ArgumentNullException(nameof(uri));
+				_directoryDocument = DocumentFile.FromSingleUri(Application.Context, uri);
+			}
+
+			internal SafFile(DocumentFile directoryDocument)
+			{
+				_directoryDocument = directoryDocument ?? throw new ArgumentNullException(nameof(directoryDocument));
+				_fileUri = _directoryDocument.Uri;
+			}
+
+			public override StorageProvider Provider => StorageProviders.AndroidSaf;
+
+			public override string Name => _directoryDocument?.Name ?? string.Empty;
```

Comment:
Sounds weird to me that a file has a `Name` but not a `Path`. You should probably just use the name as path for this particular storage provider. This would also remove the need to add a parameter-less ctor in the implementation base.